File: majorElement.java
import java.util.*;

public class majorElement {
    public static void findMajor(int[] array, int tamanho){
        int maxCount = 0, count;
        int index = -1;

        for(int i = 0; i < tamanho; i++){
            count = 0;
            for(int j = 0; j < tamanho; j++){
                if(array[i] == array[j]){
                    count++;
                }
            }
            if(count > maxCount){
                maxCount = count;
                index = i;
            }
        }
        if(maxCount > (int)tamanho/2){
            System.out.println("O valor majoritário é " + array[index] + ", que aparece " + maxCount + " vezes entre as " + tamanho + " totais.");
        }
        else{
            System.out.println("Não há valor majoritário.");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int x, a = 0;
        int[] arrayint;
        System.out.println("Defina o número (natural positivo) de elementos do array: ");
        while(a == 0) {
            try {
                x = scanner.nextInt();
                if (x <= 0) {
                    System.out.println("Erro. O número digitado é inválido.");
                    a = 1;
                }
                else{
                    arrayint = new int[x];
                    for(int i = 0; i < arrayint.length; i++){
                        System.out.println("Digite o elemento " + (i+1) + ": ");
                        arrayint[i] = scanner.nextInt();
                    }
                    System.out.println("Seu array é: " + Arrays.toString(arrayint));

                    findMajor(arrayint, arrayint.length);
                    a = 1;
                }
            } catch (Exception e) {
                System.out.println("Erro. O número digitado é inválido.");
                a = 1;
            }
        }
    }
}

File: majorElement.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
	
	int n, maxC, contagem, index = -1, x, i, j;
	
	printf("Defina o numero de elementos do vetor: ");
	scanf("%d", &n);
	printf("\n");
	
	int vetor[n];
	
	for(x = 0; x < n; x++){
		printf("Digite o valor do elemento %d: ", x+1);
		scanf("%d", &vetor[x]);
		printf("\n");
	}
	
	for(i = 0; i < n; i++){
		contagem = 0;
		for(j = 0; j < n; j++){
			if(vetor[i] == vetor[j]){
				contagem++;
			}
		}
		if(contagem > maxC){
			maxC = contagem;
			index = i;
		}
	}
	
	if(maxC > n/2){
		printf("\nO valor majoritario eh %d, que aparece %d vezes entre as %d totais.", vetor[index], maxC, n);
	}
	else{
		printf("Nao ha valor majoritario.");
	}
	
	
	return 0;
}
